### PR TITLE
add SetCapacity and GetCapacity for secondary cache

### PR DESCRIPTION
**Pull request by gitbw95**
Summary:
To support tuning secondary cache dynamically, add `SetCapacity()` and `GetCapacity()` for CompressedSecondaryCache.

Test Plan:
Unit Tests

### CODE REVIEW

File: include/rocksdb/secondary_cache.h [hunk 1]
```diff
@@ -92,7 +92,15 @@ class SecondaryCache : public Customizable {
...
   virtual void WaitAll(std::vector<SecondaryCacheResultHandle*> handles) = 0;
 
-  virtual std::string GetPrintableOptions() const = 0;
+  // Set the maximum configured capacity of the cache.
+  // When the new capacity is less than the old capacity and the existing usage
+  // is greater than new capacity, the implementation will do its best job to
+  // purge the released entries from the cache in order to lower the usage.
+  //
+  // The derived class can make this function no-op and return NotSupported().
+  virtual Status SetCapacity(size_t capacity) = 0;
```
Comment (anand1976) [inline include/rocksdb/secondary_cache.h:101]: Why do these need to be pure virtual? You can just provide default implementations that return NotSupported.

File: include/rocksdb/secondary_cache.h [hunk 2]
```diff
@@ -92,7 +92,15 @@ class SecondaryCache : public Customizable {
...
+
+  virtual size_t GetCapacity() const = 0;
```
Comment (anand1976) [inline include/rocksdb/secondary_cache.h:103]: GetCapacity should return a Status too, so the default can return NotSupported.

**Comment by gitbw95** [comment]
The comments are addressed.

**Review by anand1976 — APPROVED** [review]
LGTM

**Pull request closed without merging**